### ISSUE TITLE
Index.js is missing

### ISSUE THREAD

**Issue by gianpietro1** [sasstools/scss-tokenizer#12]
Hi,

Not sure why, my environment has been pulling a scss-tokenizer version without an index.js file, so I've had to manually generate it every time.

It's been called by saas-graph 2.1.1, which package.json file says: ""scss-tokenizer": "^0.2.1""

Details of my environment (I know it's kind of old, but it's an application requirement):
- Ubuntu Trusty 14.04.4
- Node version 0.10.25
- NPM version 1.3.10

Regards,
Gianpietro

**Comment by xzyfer** [comment]
This might be related to your old version of npm. Not sure why exactly or
how to fix it. I'll need to investigate.

**Comment by jfridye** [comment]
Thanks for checking into this. 

I'm seeing a similar error with npm v0.10.38. This started within the past few days. We have sass-graph@2.2.2 which depends on scss-tokenizer@0.2.1.

**Comment by xzyfer** [comment]
That's a node version not an npm version. Node 0.10 and npm 1 are end of
lifed so I'm not sure if there's anything we can do but I'll try to
reproduce it locally.

On 4 May 2017 12:00 am, "Joseph Fridye" <notifications@github.com> wrote:

Thanks for checking into this.

I'm seeing a similar error with npm v0.10.38. This started within the past
few days. We have sass-graph@2.2.2 which depends on scss-tokenizer@0.2.1.

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/sasstools/scss-tokenizer/issues/12#issuecomment-299048142>,
or mute the thread

**Comment by feraudet** [comment]
+1

**Comment by rasta-rocket** [comment]
+1

**Comment by xzyfer** [comment]
Can you all please confirm your node and npm versions?

On 11 May 2017 5:06 pm, "Bruno FERNANDO" <notifications@github.com> wrote:

+1

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/sasstools/scss-tokenizer/issues/12#issuecomment-300818631>,
or mute the thread

**Comment by rasta-rocket** [comment]
npm version 1.3.10
node version v0.10.25

**Comment by jfridye** [comment]
npm 1.4.28
node v0.10.38

**Comment by xzyfer** [comment]
I have not been able to reproduce this locally with 

```
➔ node -v
v0.10.48
➔ npm -v
1.4.29
```

Could y'all please show the output of listing the files in `scss-tokenizer` directory?

>ls -lsa node_modules/node-sass/node_modules/sass-graph/node_modules/scss-tokenizer

_* the path may be different on your system_.

Also please try downloading the tarball directly and confirming the `index.js` is there. You can find the tarball by running

>npm v scss-tokenizer

The end of the output should how the tarball url

```json
  dist:
   { shasum: '07c0cc577bb7ab4d08fd900185adbf4bc844141d',
     tarball: 'https://registry.npmjs.org/scss-tokenizer/-/scss-tokenizer-0.2.1.tgz' },
```

**Comment by rasta-rocket** [comment]
```
$ npm install node-sass
  ...

$ ls -lsa node_modules/node-sass/node_modules/sass-graph/node_modules/scss-tokenizer
total 24
4 drwxrwxr-x 4 user group 4096 May 12 09:53 .
4 drwxrwxr-x 5 user group 4096 May 12 09:53 ..
4 drwxr-xr-x 2 user group 4096 May 12 09:53 lib
4 drwxrwxr-x 4 user group 4096 May 12 09:53 node_modules
4 -rw-r--r-- 1 user group 2387 May 12 09:53 package.json
4 -rw-r--r-- 1 user group 1238 Jan 29 04:58 README.md

$ npm v scss-tokenizer | tail
dist: 
   { shasum: '07c0cc577bb7ab4d08fd900185adbf4bc844141d',
     tarball: 'https://registry.npmjs.org/scss-tokenizer/-/scss-tokenizer-0.2.1.tgz' },

$ wget https://registry.npmjs.org/scss-tokenizer/-/scss-tokenizer-0.2.1.tgz
--2017-05-12 13:13:14--  https://registry.npmjs.org/scss-tokenizer/-/scss-tokenizer-0.2.1.tgz
Resolving registry.npmjs.org (registry.npmjs.org)... 151.101.120.162
Connecting to registry.npmjs.org (registry.npmjs.org)|151.101.120.162|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 5777 (5.6K) [application/octet-stream]
Saving to: ‘scss-tokenizer-0.2.1.tgz’

100%[================================================================================================================================================================================================>] 5,777       --.-K/s   in 0s      

2017-05-12 13:13:14 (70.0 MB/s) - ‘scss-tokenizer-0.2.1.tgz’ saved [5777/5777]

$ tar -xzf scss-tokenizer-0.2.1.tgz

$ ls package/
total 20
-rw-r--r-- 1 user group   49 Apr 29 01:52 index.js
drwxrwxr-x 2 user group 4096 May 12 13:13 lib
-rw-r--r-- 1 user group 1077 Jan 28 11:17 LICENSE
-rw-r--r-- 1 user group  963 Apr 29 02:14 package.json
-rw-r--r-- 1 user group 1238 Jan 29 04:58 README.md
 ```

**Comment by xzyfer** [comment]
Thanks @rasta-rocket, this is facinating.

**Comment by xzyfer** [comment]
Ok I've been able to reproduce this with 

```
➔ node -v
v0.10.25
➔ npm -v
1.3.10
```

After some npm version bisecting I have been able to reproduce this issue up until npm@1.4.0 but not after. This conflicts with the information provided by @jfridye.

I'm going to investigate some potential workarounds

**Comment by xzyfer** [comment]
I believe this is resolved in v0.2.3. Please confirm.

**Comment by rasta-rocket** [comment]
Also I tested the 0.2.0 version, the index.js is present

**Comment by xzyfer** [comment]
Fixed in d84958c

**Comment by rasta-rocket** [comment]
Thx. Do you will update the new version on sass-graph package.json ?

**Comment by xzyfer** [comment]
Published sass-graph@2.2.3
